osgi.enroute.web.simple.provider/src/osgi/enroute/web/server/provider/WebServer.java

```diff
@@ -663,8 +636,13 @@ Cache findBundle(String path) throws Exception {
 		if (bundles != null) {
 			for (Bundle b : bundles) {
 				Enumeration<URL> urls = b.findEntries("static/" + path, "*", false);
+				// What happens here is that we have hit a folder, but the path does not
+				// end with a "/". I do not think that it is correct to do a redirect here.
+				// In any case, when redirects are turned off, this causes an infinite redirect loop.
+				// Instead, a 404 should be thrown.
+				// I would argue that a 404 should **always** be thrown here for this case.
 				if (urls != null && urls.hasMoreElements()) {
-					throw new RedirectException("/" + path + "/index.html");
```

Comment:
As I try to explain in the comment, this is making a *huge* assumption.

Perhaps not everybody wants to treat this like a file system, like Apache does by default. I think in the current age, a web server is much more than just a file server, so we don't necessary need to assume that a "folder" should act like a directory and print an "index".

If the backend implementation of a URI does happen to be a folder, then perhaps some other action is required, like return a 404, or an error, or some other kind of message. (In my case, I want a 404.)

I actually believe that it is *not* proper HTTP to arbitrarily append a "/" at the end of a URI. That is something that Apache did to try to be "nice" when showing files on a filesystem, and somehow that became the norm. I don't think it is the right way to act, so I would argue that returning a 404 is correct. However, I understand that not everybody may think that way, so it should be configurable.

That said, redirecting when the URI ends with "/", and redirecting when a URI (that does not end with "/") happens to be a "folder" are two different things, IMO.